fgpyo/fasta/reference_set_builder.py

```diff
@@ -0,0 +1,174 @@
+"""
+Classes for generating Fasta files and records for testing
+----------------------------------------------------------------
+"""
+# import hashlib
+import os
+import textwrap
+from tempfile import NamedTemporaryFile
+from typing import List
+from typing import Optional
+
+# pylint: disable=W0511
+
+
+class ReferenceSetBuilder:
+    """
+    Builder for constructing one or more fasta records.
+    """
+
+    # The default asssembly
+    DEFAULT_ASSEMBLY: str = "testassembly"
+
+    # The default species
+    DEFAULT_SPECIES: str = "testspecies"
+
+    # Way to store instance of ReferenceBuilder
+    # TODO make something better than a list... probably
+    REF_BUILDERS: List["ReferenceBuilder"] = []
+
+    def __init__(
+        self,
+        assembly: Optional[str] = None,
+        species: Optional[str] = None,
+        line_length: int = 80,
+    ):
+        self.assembly: str = assembly if assembly is not None else self.DEFAULT_ASSEMBLY
+        self.species: str = species if species is not None else self.DEFAULT_SPECIES
+        self.line_length = line_length
+
+    def add(
+        self,
+        name: str,
+    ) -> "ReferenceBuilder":
+        """
```

Comment:
add args to docstring here and below

Reply:
Added docstrings to all classes and functions, see commit cb8ac832a05ba4a3ad304edb2de19c57d6f9b8f3